fearless_simd/src/core_arch/x86/avx.rs

```diff
@@ -0,0 +1,336 @@
+// Copyright 2024 the Fearless_SIMD Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![expect(
+    clippy::too_many_arguments,
+    reason = "TODO: https://github.com/linebender/fearless_simd/issues/40"
+)]
```

Comment:
I'm surprised that this expect is required. We've removed too_many_arguments from the lint set, and so all expects of it should have been removed at the same time